### PR TITLE
599 fix cob measurement noise

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** MAXGNC-599
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The computation of the COB error covariance was updated such that the measurement noise increases with number of pixels, instead of decreasing. Additionally, the covariance of the attitude uncertainty was included to compute the total covariance matrix.

## Verification

The UnitTest was updated.

## Documentation

The documentation was updated.

## Future work

N/A

### CODE REVIEW

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.rst [hunk 1]
```diff
@@ -75,15 +75,17 @@ where :math:`\mathbf{\bar{u}}_{COB} = [\mathrm{cob}_x, \mathrm{cob}_y, 1]^T` wit
 brightmess :math:`\mathrm{cob}_x` and :math:`\mathrm{cob}_y`, :math:`[K]` is the camera calibration matrix and
 :math:`\mathbf{r}_{COB}^N` is the unit vector describing the physical heading to the target in the camera frame.
 
-The covariance is found using the number of detected pixels and the camera parameters, given by:
+The covariance of the COB error is found using the number of detected pixels and the camera parameters, given by:
 
 .. math::
 
-    P = \frac{2 \ pi}{\sqrt{\mathrm{numPixels}}} \left( \begin{bmatrix} d_x^2 & 0 & 0 \\ 0 & d_y^2 & 0
+    P = \frac{\sqrt{\mathrm{numPixels}}}{2 \pi} \left( \begin{bmatrix} d_x^2 & 0 & 0 \\ 0 & d_y^2 & 0
```
Comment (thibaudteil) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.rst]: Is this 4/pi divided by the norm?

**Review by thibaudteil — APPROVED** [review]
Just a documentation comment

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.rst (hunk 1 above)
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.rst]: Yes, good catch. I forgot to update the documentation after we changed that computation again.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp [hunk 2]
```diff
@@ -154,17 +154,21 @@ void CobConverter::UpdateState(uint64_t CurrentSimNanos)
...
 
-        /*! - Define diagonal terms of the covariance */
-        Eigen::Matrix3d covar_C;
-        covar_C.setZero();
-        covar_C(0,0) = pow(X,2);
-        covar_C(1,1) = pow(Y,2);
-        covar_C(2,2) = 1;
-        /*! - define and rotate covariance using number of pixels found */
+        /*! - define diagonal terms of the COB covariance */
+        Eigen::Matrix3d covar_cob_C;
```
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: :large_orange_diamond: Variable name doesn't really adhere to code guidelines. Perhaps, `covarCob_C` or similar is better? Happy to discuss of course.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp [hunk 3]
```diff
@@ -217,7 +221,7 @@ double CobConverter::getRadius() const {
     return this->objectRadius;
 }
 
-/*! Set the attitude error covariance matrix in body frame B
+/*! Set the attitude error covariance matrix in body frame B, for unit vector measurements
```
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp:224]: :large_orange_diamond: Should be squashed into the first commit.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp (hunk 2 above)
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp]: Fixed it in the corresponding commit. Also included an additional [Refactor] commit at the end to fix the naming of older variables to adhere to code guidelines.

File: src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp (hunk 3 above)
Comment (JulianHammerl) [reply src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp:224]: Done.

**Pull request merged**